dashboard/app/views/shared/_course_tall_block.haml

```diff
@@ -37,95 +38,83 @@
   - url = CDO.code_org_url('/hourofcode/unplugged-conditionals-with-cards')
   - title = t('upsell.unplugged_conditionals.title')
   - body = t('upsell.unplugged_conditionals.body')
+  - teacher_guide = CDO.code_org_url("/hourofcode/unplugged-conditionals-with-cards")
 
-- elsif id == 'sports'
+- elsif id == Script::SPORTS_NAME
   - url = CDO.code_org_url('/athletes')
   - title = t('upsell.sports.title')
   - body = t('upsell.sports.body')
 
-- elsif id == 'text_compression'
+- elsif id == Script::TEXT_COMPRESSION_NAME
   - url = CDO.studio_url('/s/text-compression/reset')
   - title = t('upsell.text-compression.title')
   - body = t('upsell.text-compression.body')
+  - teacher_guide = "https://curriculum.code.org/hoc/2016/1/"
 
-- elsif id == 'hoc_encryption'
+- elsif id == Script::HOC_ENCRYPTION_NAME
   - url = CDO.studio_url('/s/hoc-encryption/reset')
   - title = t('upsell.hoc-encryption.title')
   - body = t('upsell.hoc-encryption.body')
+  - teacher_guide = "https://curriculum.code.org/hoc/2016/2/"
 
 - elsif [Script::COURSE1_NAME, Script::COURSE2_NAME,Script::COURSE3_NAME, Script::COURSE4_NAME, Script::COURSEA_NAME, Script::COURSEB_NAME, Script::COURSEC_NAME, Script::COURSED_NAME, Script::COURSEE_NAME, Script::COURSEF_NAME].include? id
   - url = script_url(id)
   - title = data_t_suffix('script.name', id, 'title')
   - body = data_t_suffix('script.name', id, 'description_short')
+  - audience = data_t_suffix('script.name', id, 'description_audience')
 
 - elsif [Script::ARTIST_NAME, Script::PLAYLAB_NAME, Script::HOC_NAME, Script::FROZEN_NAME, Script::INFINITY_NAME, Script::STARWARS_NAME, Script::STARWARS_BLOCKS_NAME, Script::MINECRAFT_NAME, Script::MINECRAFT_DESIGNER_NAME, Script::MINECRAFT_HERO_NAME, Script::MINECRAFT_AQUATIC_NAME, Script::DANCE_PARTY_NAME].include? id
   - url = script_reset_url(id)
-  - if id == 'frozen'
+  - body = data_t_suffix('script.name', id, 'description_short')
+  - if id == Script::FROZEN_NAME
```

Comment:
I wonder if it'd be good to replace this big set of conditionals with an array of hashes?

Reply:
It would probably also be good to get this ruby into a `:ruby` block at minimum, if not a `.rb` file.

Reply:
yeah, i think you're right. i'll do that in a follow-up PR